code-orange/ViewController.swift

```diff
@@ -9,17 +9,50 @@
 import UIKit
 import GoogleMaps
 
-protocol CurrentLocationProvider: class {
+protocol LocationProvider: class {
   var currentLocation: CLLocationCoordinate2D { get }
 }
 
+protocol CoordinatesProviderDelegate: class {
+  func coordinatesProviderUpdated(_ sender: CoordinatesProvider)
+}
+
+protocol CoordinatesProvider: class {
+  var coordinates: [CLLocationCoordinate2D] { get }
+  var delegate: CoordinatesProviderDelegate? { get set }
+}
+
 class ViewController: UIViewController {
```

Comment:
Can we give this task a meaningful name?

Reply:
Done 🚀